src/main/java/org/elasticsearch/action/admin/cluster/repositories/verify/TransportVerifyRepositoryAction.java

```diff
@@ -69,7 +69,7 @@ protected VerifyRepositoryResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(VerifyRepositoryRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
Verification is called in two cases - when we create repo and when we explicitly call verification. We already check for `METADATA_WRITE` in the put repository operation, so I think we can use `METADATA_READ` here.